resolver.go

```diff
@@ -501,6 +544,54 @@ func (r *resolver) forwardQueryStart(w dns.ResponseWriter, msg *dns.Msg, queryID
 	return true
 }
 
+func (r *resolver) addQueryToGC(w dns.ResponseWriter, msg *dns.Msg) {
+	var (
+		cc clientConn
+		ok bool
+	)
+	proto := w.LocalAddr().Network()
+	if proto != "udp" {
+		return
+	}
+
+	r.queryLock.Lock()
+	if cc, ok = r.client[msg.Id]; !ok {
+		return
+	}
+	r.queryLock.Unlock()
+
+	query := sboxQuery{
+		sboxID: r.sb.ID(),
+		dnsID:  msg.Id,
+	}
```

Comment:
Seems like sboxQuery is only used as a unique id for a query on a specific sbox.
a string would do it and we can avoid to construct yet another struct.

`fmt.Sprintf("%s-%d",r.sb.ID(), msg.Id)` 

Up to you.